examples/examples_test.go

```diff
@@ -29,3 +35,88 @@ func skipIfShort(t *testing.T) {
 		t.Skip("skipping long-running test in short mode")
 	}
 }
+
+func assertHTTPResult(t *testing.T, output interface{}, headers map[string]string, check func(string) bool) bool {
+	return assertHTTPResultWithRetry(t, output, headers, 5*time.Minute, check)
+}
+
+func assertHTTPResultWithRetry(t *testing.T, output interface{}, headers map[string]string, maxWait time.Duration, check func(string) bool) bool {
+	return assertHTTPResultShapeWithRetry(t, output, headers, maxWait, func(string) bool { return true }, check)
+}
+
+func assertHTTPResultShapeWithRetry(t *testing.T, output interface{}, headers map[string]string, maxWait time.Duration,
+	ready func(string) bool, check func(string) bool) bool {
+	hostname, ok := output.(string)
+	if !assert.True(t, ok, fmt.Sprintf("expected `%s` output", output)) {
+		return false
+	}
+
+	if !(strings.HasPrefix(hostname, "http://") || strings.HasPrefix(hostname, "https://")) {
+		hostname = fmt.Sprintf("http://%s", hostname)
+	}
+
+	startTime := time.Now()
+	count, sleep := 0, 0
+	for true {
+		now := time.Now()
+		req, err := http.NewRequest("GET", hostname, nil)
+		if !assert.NoError(t, err) {
+			return false
+		}
+
+		for k, v := range headers {
+			// Host header cannot be set via req.Header.Set(), and must be set
+			// directly.
+			if strings.ToLower(k) == "host" {
+				req.Host = v
+				continue
+			}
+			req.Header.Set(k, v)
+		}
+
+		client := &http.Client{Timeout: time.Second * 10}
+		resp, err := client.Do(req)
+		if err == nil && resp.StatusCode == 200 {
+			if !assert.NotNil(t, resp.Body, "resp.body was nil") {
+				return false
+			}
+
+			// Read the body
+			defer resp.Body.Close()
+			body, err := ioutil.ReadAll(resp.Body)
+			if !assert.NoError(t, err) {
+				return false
+			}
+
+			bodyText := string(body)
+
+			// Even if we got 200 and a response, it may not be ready for assertion yet - that's specific per test.
+			if ready(bodyText) {
+				// Verify it matches expectations
+				return check(bodyText)
+			}
+		}
+		if now.Sub(startTime) >= maxWait {
+			fmt.Printf("Timeout after %v. Unable to http.get %v successfully.", maxWait, hostname)
+			return false
+		}
+		count++
+		// delay 10s, 20s, then 30s and stay at 30s
+		if sleep > 30 {
```

Comment:
Maybe just replace with "github.com/jpillora/backoff" as we do in the provider?

Reply:
I copied this 1:1 from the examples repo. We could improve both I guess.